pylint/checkers/base.py

```diff
@@ -512,13 +512,29 @@ def visit_functiondef(self, node):
                            retnode.value.value is not None:
                         self.add_message('return-arg-in-generator', node=node,
                                          line=retnode.fromlineno)
-        # Check for duplicate names
-        args = set()
-        for name in node.argnames():
-            if name in args:
-                self.add_message('duplicate-argument-name', node=node, args=(name,))
+        # Check for duplicate names by clustering args with same name for detailed report
+        args = node.args
+        arg_clusters = collections.defaultdict(list)
+        for arg in itertools.chain.from_iterable(pack for pack in
+                                                 [args.args, [args.vararg],
+                                                  args.kwonlyargs, [args.kwarg]]
+                                                 if pack is not None and all(pack)):
+            if isinstance(arg, str):
+                # sadly, asteroid doesn't expose variadic and kw-variadic arg nodes
```

Comment:
Can you elaborate? What is the currently observed behaviour? What would be a desired behaviour?

Reply:
@rogalski, thx for answers!

Currently pylint don't reports about each argument, which have duplicates. It reports only once regardless count of errors.
For example, we have this function:
```
def foo(bar, bar2, bar, *, bar=None):
    pass
```
I essentially expect that linter will generate error reports at least about two last duplicated `bar` arguments.

Reply:
I would say that we shouldn't handle variadic parameters, so basically this isinstance check could go away.